persist/fs/persist_manager.go

```diff
@@ -57,37 +105,51 @@ type persistManager struct {
 
 // NewPersistManager creates a new filesystem persist manager
 func NewPersistManager(opts Options) persist.Manager {
+	scope := opts.InstrumentOptions().MetricsScope().SubScope("persist")
 	filePathPrefix := opts.FilePathPrefix()
 	writerBufferSize := opts.WriterBufferSize()
 	blockSize := opts.RetentionOptions().BlockSize()
 	newFileMode := opts.NewFileMode()
 	newDirectoryMode := opts.NewDirectoryMode()
 	writer := NewWriter(blockSize, filePathPrefix, writerBufferSize, newFileMode, newDirectoryMode)
-	return &persistManager{
+	pm := &persistManager{
 		opts:           opts,
+		scope:          scope,
 		filePathPrefix: filePathPrefix,
 		rateLimitOpts:  opts.RateLimitOptions(),
 		nowFn:          opts.ClockOptions().NowFn(),
 		sleepFn:        time.Sleep,
 		writer:         writer,
+		metrics:        make(map[uint32]*shardMetrics),
 		segmentHolder:  make([]checked.Bytes, 2),
 	}
+
+	go pm.reportLoop()
```

Comment:
Possibly, we'd only stop it when the database is shutting down, so stopping this report loop would require threading the closing logic from the very top-level down to the persistence manager. I could do that if you feel strongly?